src/cli/subcommand/install.rs

```diff
@@ -80,21 +80,26 @@ impl CommandExecute for Install {
             false => None,
         };
 
+        let uninstall_command = match Path::new("/nix/nix-installer").exists() {
+            true => "/nix/nix-installer uninstall".into(),
+            false => format!("curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix/tag/v{} | sh -s -- uninstall", env!("CARGO_PKG_VERSION")),
```

Comment:
I don't know if this is quite correct (even excluding `-unreleased` versions, whose semantics I'm unsure of) -- if the user was trying to run `nix-installer` 0.7.0, but the receipt is 0.6.0, this won't work, right? It'll just suggest they use the same version they already are...

But I don't know a satisfying way to resolve this. Maybe say "please get a v${version from receipt} `nix-installer` from (our repo, or wherever they originally got it)"?

Reply:
Nix-installer 0.7.0 can't parse a 0.6.0 receipt :) It'd error earlier and tell them to run the right version.

Reply:
OK, I see.

Then, this is probably something for another PR, but the scenario I was imagining was this:

1. User installs with `nix-installer` v0.6.0.
2. Something happens, and `/nix/nix-installer` gets removed, but everything else stays around
3. Something else happens and they decide their best course of action is to reinstall
4. The user downloads the latest `nix-installer`, v0.7.0, via our `curl | sh` recommended pipeline
5. Nothing would work (`install` or `uninstall`), since the current receipt is 0.6.0, but the binary they're attempting to run is 0.7.0

I think at this point, if they try to run anything and rightfully get a semver mismatch error, we should _also_ suggest how to get the matching installer at that point.

Reply:
Users would get a message like these right now: https://github.com/DeterminateSystems/nix-installer/blob/b7d7afd02e9a4677da5fb017a9877d790c137543/src/plan.rs#L352-L359

Reply:
Ah, nice, perfect.